### DISCUSSION TITLE
Support for Azure SQL in Keycloak

### DISCUSSION THREAD

**Discussion by Florzl** [keycloak/keycloak#36089] [Q&A] [answered]
Hello Keycloak Community,

I’m currently evaluating the use of Azure SQL as the database for Keycloak. Before committing to this approach, I’d like to gather any experiences or insights you may have.

• Has anyone successfully deployed Keycloak with Azure SQL as the primary database in production? If so, were there any compatibility issues or additional configurations required?

• Azure SQL does not support XA transactions. Would this limitation affect Keycloak’s stability or reliability when using Azure SQL? If so, what alternatives or workarounds are recommended?

• If you’ve tested this setup, could you share any best practices or considerations? Are there specific performance, scalability, or maintenance aspects to be aware of?

I appreciate any guidance, tips, or real-world experiences you can provide.

Best regards,
Florian

**Comment by sschu** [comment, marked as answer]
We have been running Keycloak on Azure SQL for several years now and we did not encounter any problems specific to Keycloak. The lack of XA support probably means you cannot use distributed transactions but we don't need that anyways.
The problems we had where all related to temporary network issues or issues in Azure SQL itself.
